src/piratechain/PiratechainTools.ts

```diff
@@ -131,11 +130,18 @@ export class PiratechainTools implements EdgeCurrencyTools {
     if (typeof mnemonic !== 'string') {
       throw new Error('InvalidMnemonic')
     }
-    const unifiedViewingKey: UnifiedViewingKey =
-      await this.nativeTools.deriveViewingKey(
+
+    let unifiedViewingKey: string
+    try {
+      unifiedViewingKey = await this.nativeTools.deriveViewingKey(
         mnemonic,
         this.networkInfo.rpcNode.networkName
       )
+    } catch (e) {
+      // Return a failsafe value if derivation fails (e.g., native code for platform not implemented)
+      unifiedViewingKey = 'failed_to_call_deriveViewingKey'
```

Comment:
In the previous commit, I talked about leaving the birthday height in the public keys, but making it `asOptional`. What if we do the same thing to the `publicKey`? That way we can return `undefined` here, instead of some weird string "failed_to_call_deriveViewingKey".